src/galileo/logger/logger.py

```diff
@@ -134,35 +135,52 @@ def __init__(
         project_name_from_env = getenv("GALILEO_PROJECT", DEFAULT_PROJECT_NAME)
         log_stream_name_from_env = getenv("GALILEO_LOG_STREAM", DEFAULT_LOG_STREAM_NAME)
 
+        project_id_from_env = getenv("GALILEO_PROJECT_ID")
+        log_stream_id_from_env = getenv("GALILEO_LOG_STREAM_ID")
+
         self.project_name = project or project_name_from_env
+        self.project_id = project_id or project_id_from_env
 
-        if not self.project_name:
+        if self.project_name is None and self.project_id is None:
             raise GalileoLoggerException(
-                "User must provide project_name to GalileoLogger, or set it as an environment variable."
+                "User must provide project_name or project_id to GalileoLogger, or set it as an environment variable."
             )
 
-        if log_stream and experiment_id:
-            raise GalileoLoggerException("User must provide either experiment_id or log_stream, not both.")
+        if (log_stream or log_stream_id) and experiment_id:
+            raise GalileoLoggerException("User cannot specify both a log stream and an experiment.")
```

Comment:
This validation should occur after L156, no? Until then, we don't really _know_ if we have been given both.